From 8fa7bdf8d1ec239751d2d27ce390b791f341fe07 Mon Sep 17 00:00:00 2001
From: David Teigland <teigland@redhat.com>
Date: Mon, 23 Mar 2026 17:10:56 -0500
Subject: [PATCH 2/4] lvmlockd: fix building for dlm

when LOCKDSANLOCK_SUPPORT is not defined.

(cherry picked from commit f6128ac29b53bedeb6ff6c7dcd3bb777afd2a401)
---
 daemons/lvmlockd/lvmlockd-core.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/daemons/lvmlockd/lvmlockd-core.c b/daemons/lvmlockd/lvmlockd-core.c
index d11cd40db..41a734a4d 100644
--- a/daemons/lvmlockd/lvmlockd-core.c
+++ b/daemons/lvmlockd/lvmlockd-core.c
@@ -7384,7 +7384,11 @@ static int main_loop(daemon_state *ds_arg)
 	pthread_mutex_init(&log_mutex, NULL);
 
 	openlog("lvmlockd", LOG_CONS | LOG_PID, LOG_DAEMON);
+#ifdef LOCKDSANLOCK_SUPPORT
 	log_warn("lvmlockd started " LVM_VERSION " sanlock_support %u", LOCKDSANLOCK_SUPPORT);
+#else
+	log_warn("lvmlockd started " LVM_VERSION " sanlock_support no");
+#endif
 
 	listen_fd = ds_arg->socket_fd;
 	listen_pi = add_pollfd(listen_fd);
@@ -7619,7 +7623,11 @@ int main(int argc, char *argv[])
 			usage(argv[0], stdout);
 			exit(EXIT_SUCCESS);
 		case 'V':
+#ifdef LOCKDSANLOCK_SUPPORT
 			printf("lvmlockd version: " LVM_VERSION " sanlock_support %u\n", LOCKDSANLOCK_SUPPORT);
+#else
+			printf("lvmlockd version: " LVM_VERSION " sanlock_support no\n");
+#endif
 			exit(EXIT_SUCCESS);
 		case 'T':
 			daemon_test = 1;
-- 
2.53.0

